src/client/auth/auth.go

```diff
@@ -36,3 +39,19 @@ func (e NotActivatedError) Error() string {
 func IsNotActivatedError(e error) bool {
 	return strings.Contains(e.Error(), notActivatedErrorMsg)
 }
+
+// In2Out converts an incoming context containing auth information into an
+// outgoing context containing auth information, stripping other keys (e.g.
+// for metrics) in the process. If the incoming context doesn't have any auth
+// information, then the returned context won't either.
+func In2Out(ctx context.Context) context.Context {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return ctx
+	}
+	mdOut := make(metadata.MD)
+	if value, ok := md[ContextTokenKey]; ok {
+		mdOut[ContextTokenKey] = value
+	}
+	return metadata.NewOutgoingContext(ctx, mdOut)
+}
```

Comment:
Is this function doing anything besides stripping out extraneous information like metrics? If so what? It seems like the metadata library distinguishes between incoming and outgoing messages but I can't quite understand to what end it does this.